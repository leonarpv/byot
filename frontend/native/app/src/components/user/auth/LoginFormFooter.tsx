import * as React from 'react';
import {StyleSheet, View} from 'react-native';
import {useTranslation} from 'react-i18next';
import {Button} from '../../elements/lib/Button';

interface Props {
  onPasswordRequestPress: () => void;
  onRegisterPress: () => void;
}

interface State {}

const makeStyles = (props: Props, state: State) =>
  StyleSheet.create({
    root: {
      flexDirection: 'row',
      justifyContent: 'space-between',
      marginTop: 10,
    },
  });

export const LoginFormFooter: React.FC<Props> = (props: Props) => {
  const styles = makeStyles(props, {});
  const {t} = useTranslation();
  const testIdForgotPassword = 'auth-loginFormFooter-forgotPassword';
  const testIdRegister = 'auth-loginFormFooter-register';
  return (
    <View style={styles.root}>
      <Button
        onPress={props.onPasswordRequestPress}
        variant="secondary"
        testID={testIdForgotPassword}
        link
        label={t('Forgot password?')}
      />
      <Button
        onPress={props.onRegisterPress}
        variant="primary"
        testID={testIdRegister}
        link
        label={t(`Don't have an account?`)}
      />
    </View>
  );
};
