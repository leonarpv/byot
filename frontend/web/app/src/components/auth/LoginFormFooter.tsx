import * as React from 'react';
import {Box} from '@material-ui/core';
import {Link} from 'react-router-dom';
import {Button} from '@byot-frontend/web-common/src/components/elementary/form/Button';
import {Router} from '../../router/Router';
import {useTranslation} from 'react-i18next';

interface Props {}

export const LoginFormFooter: React.FC<Props> = (props: Props) => {
  const {t} = useTranslation();
  return (
    <Box display="flex" flexDirection="row" justifyContent="space-between" mt={4}>
      <Button variant="text" component={Link} to={Router.resetPassword.URI()}>
        {t('Forgot password?')}
      </Button>
      <Button color="secondary" variant="text" component={Link} to={Router.register.URI()}>
        {t("Don't have an account?")}
      </Button>
    </Box>
  );
};
