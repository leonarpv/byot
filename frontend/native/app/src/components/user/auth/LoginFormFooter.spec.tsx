import * as React from 'react';
import {render, fireEvent, wait} from '@testing-library/react-native';
import {LoginFormFooter} from './LoginFormFooter';

describe('<LoginFormFooter/>', () => {
  it('should render and call events', async () => {
    const onRegisterPress = jest.fn();
    const onPasswordResetPress = jest.fn();
    const {asJSON, getByTestId} = render(
      <LoginFormFooter onRegisterPress={onRegisterPress} onPasswordRequestPress={onPasswordResetPress} />
    );
    fireEvent.press(getByTestId('auth-loginFormFooter-forgotPassword'));
    await wait(() => expect(onPasswordResetPress).toBeCalled());
    fireEvent.press(getByTestId('auth-loginFormFooter-register'));
    await wait(() => expect(onRegisterPress).toBeCalled());
    expect(asJSON()).toMatchSnapshot();
  });
});
