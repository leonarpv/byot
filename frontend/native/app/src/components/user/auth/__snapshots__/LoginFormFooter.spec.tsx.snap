// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginFormFooter/> should render and call events 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 10,
      }
    }
  >
    <TouchableOpacity
      accessibilityLabel="auth-loginFormFooter-forgotPassword"
      label="Forgot password?"
      link={true}
      testID="auth-loginFormFooter-forgotPassword"
      variant="secondary"
    />
    <TouchableOpacity
      accessibilityLabel="auth-loginFormFooter-register"
      label="Don't have an account?"
      link={true}
      testID="auth-loginFormFooter-register"
      variant="primary"
    />
  </View>
</View>
`;
